SingularityS3Uploader/src/main/java/com/hubspot/singularity/s3uploader/SingularityUploader.java

```diff
@@ -68,6 +72,8 @@
     this.logIdentifier = String.format("[%s]", metadataPath.getFileName());
     this.configuration = configuration;
     this.exceptionNotifier = exceptionNotifier;
+
+    this.checkFileOpenLock = new ReentrantLock();
```

Comment:
Just checking, is this class always a singleton? Want to make sure we aren't just creating multiple locks. I had thought the singleton was the uploader driver, not the uploader

Reply:
Good catch. Fixed.